BranchMetrics/ios-branch-deep-linking-attribution
# [SDK-2035] Remove unnecessary fields from v1/url requests

## Reference
SDK-2035 -- Remove unnecessary fields from v1/url requests

## Summary
Added a section of code to `preparePostRequest()` in `BNCServerInterface.m` which removes 10 fields from `v1/url` requests which are no longer necessary. This change also adds a constant for `anon_id`.

## Motivation

The motivation for this change is to reduce the number of fields in v1/url requests down to only what's required or used. 

## Type Of Change


## Testing Instructions

Create short url's and observe the outgoing requests. They should not contain any of the 10 removed fields. 
```
SESSION_ID, 
OPTED_IN_STATUS, 
RANDOMIZED_DEVICE_TOKEN, 
RANDOMIZED_BUNDLE_TOKEN, 
IS_HARDWARE_ID_REAL, 
IOS_VENDOR_ID, 
HARDWARE_ID_TYPE, 
HARDWARE_ID, 
AD_TRACKING_ENABLED, 
ANON_ID
```









cc @BranchMetrics/saas-sdk-devs for visibility.
